src/renderer/details/components/ShipNameField.tsx

```diff
@@ -9,9 +13,15 @@ interface ShipNameFieldProps {
 }
 
 export const ShipNameField: React.FC<ShipNameFieldProps> = ({ form, validator, placeholder = '~sampel-palnet' }) => {
+    const { data: piers } = useQuery(pierKey(), () => send('get-piers'))
     const shipnamePattern = /^[a-z~-]*$/i;
     const shipnameContainsInvalidCharacters = form.errors.shipName?.type === 'pattern';
 
+    function getMatchingPier(shipName: string): Pier | undefined {
+        return piers.find(pier => pier.shipName.trim().toLowerCase() === shipName.toLowerCase())
```

Comment:
`shipName` can possibly be undefined here for a remote ship

```suggestion
        return piers.find(pier => pier.shipName?.trim().toLowerCase() === shipName.toLowerCase())
```